src/support.cpp

```diff
@@ -48,6 +48,261 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+    {
+        assert(storage.support.supportLayers[layer_nr].support_infill_parts.empty() && "support infill part list is supposed to be uninitialized");
+
+        // this is the complete support areas on this layer
+        const Polygons& whole_support_areas = storage.support.supportLayers[layer_nr].supportAreas;
+
+        if (whole_support_areas.size() == 0 || layer_nr < min_layer || layer_nr > max_layer)
+        {
+            // initialize support_infill_parts empty
+            storage.support.supportLayers[layer_nr].support_infill_parts.clear();
+            continue;
+        }
+
+        // generate separate support islands and calculate density areas for each island
+        std::vector<PolygonsPart> support_islands = whole_support_areas.splitIntoParts();
+        for (unsigned int i = 0; i < support_islands.size(); ++i)
+        {
+            storage.support.supportLayers[layer_nr].support_infill_parts.emplace_back();
+            SupportInfillPart& support_infill_part = storage.support.supportLayers[layer_nr].support_infill_parts.back();
+            support_infill_part.gradual_infill_areas_per_combine_per_density.clear();
+            support_infill_part.outline = support_islands[i];
+            support_infill_part.insets.clear();
+            // generate insets and use the first inset as the infill area
+            AreaSupport::generateOutlineInsets(
+                support_infill_part.insets,
+                support_infill_part.outline,
+                wall_line_count,
+                support_line_width);
+            assert(support_infill_part.gradual_infill_areas_per_combine_per_density.size() == 0);
+
+            support_infill_part.inner_infill_areas.clear();
+            if (support_infill_part.insets.empty())
+            {
+                continue;
+            }
+            support_infill_part.inner_infill_areas = support_infill_part.insets[0];
+
+            // calculate density areas for this island
+            Polygons less_dense_support = support_infill_part.inner_infill_areas; // one step less dense with each density_step
+            for (unsigned int density_step = 0; density_step < max_density_steps; ++density_step)
+            {
+                size_t min_layer = layer_nr + density_step * gradual_support_step_layer_count + layer_skip_count;
+                size_t max_layer = layer_nr + (density_step + 1) * gradual_support_step_layer_count;
+
+                for (float upper_layer_idx = min_layer; static_cast<unsigned int>(upper_layer_idx) <= max_layer; upper_layer_idx += layer_skip_count)
+                {
+                    if (static_cast<unsigned int>(upper_layer_idx) >= total_layer_count)
+                    {
+                        less_dense_support.clear();
+                        break;
+                    }
+                    const Polygons& relevent_upper_polygons = storage.support.supportLayers[upper_layer_idx].supportAreas;
```

Comment:
You are comparing `inner_infill_areas` with the whole `supportAreas`.
This is rather surprising to me.

I had expected you to first process the insets separately and then only look at the `inner_infill_areas` only in this function.

Comparing with the whole `supportAreas` yields different results, but they might be better. This is actually the very reason I implemented the Skin Reference Wall feature.

Let me explain:
```
legend:
^ support roof
| support wall
# dense support
+ less dense support

comparing infill    comparing with outline
   ^^^^^^           ^^^^^^
   ####||^^         ####||^^
   ######||^^       ######||^^
   ++++####||       ++++++##||
   ++++++####       ++++++++##
```

I'm not a 100% sure, but I think the second method is better.
Whichever you use, please document clearly why you chose that way of implementing it.